docs/tutorials/time-series.md

```diff
@@ -0,0 +1,766 @@
+data with Feldera.  A time series is a sequence of events, such as IoT sensor
+readings or financial transactions, where each event is associated with one or
+more timestamps.  Time series data is inherently dynamic, changing continuously
+as new events arrive in real-time.  Feldera is designed to
+efficiently compute over changing data, making it well-suited for time series
+analytics.
+
+:::tip
+
+Most examples in this guide are available as a pre-packaged demo "Time Series Analysis with Feldera"
+that ships with the
+[Feldera Docker container](https://docs.feldera.com/docker) and the
+[online sandbox](https://try.feldera.com).
+
+:::
+
+**Time series data is not special**.  The most important thing to know about
+time series processing in Feldera is that for most intents and purposes time
+series are handled in the same way as any other input data, in particular:
+
+  * Time series inputs are modeled as SQL tables with [input connectors](/connectors)
+    attached to them.
+
+  * Time series tables support inserts, deletes, and upserts.
+
+  * **All** SQL operators apply to time series and non-time-series data with
+    identical semantics, i.e., given the same inputs, Feldera produces the exact
+    same outputs.
+
+**Feldera optimizes time series processing**.  While Feldera offers the same
+expressive power for time series and non-time-series data, it can evaluate some
+classes of time series queries faster and/or using less memory:
+
+  * It takes advantage of the fact that timestamps in a time series grow monotonically or
+    almost monotonically.  This allows discarding old data no longer needed to compute outputs
+    for new data points. In order to apply this optimization the query engine needs
+    to know which columns in the input time series contain (almost) monotonically growing
+    timestamp values.  This requires
+    [`LATENESS` annotations](#timestamp-columns-and-lateness), described below.
+
+  * Time series data is often append-only: once an event has been appended to the
+    series, it cannot be deleted or modified.  Feldera implements an optimized version
+    of several operators for append-only collections.  In order to enable these
+    optimizations, the query engine needs to know which inputs are append-only.  The user
+    can provide this information to the engine using [`append_only` annotations](#append-only-tables).
+
+As a user, you do not need to understand the intricate details of time series
+processing in order to take advantage of these optimizations. Your simply
+communicate information about the input data to the query engine using
+annotations.  Feldera uses this information to apply optimizations automatically
+in a way that does not affect the output of the queries.  **Assuming that user
+annotations are correct, the optimized query plan produces the same outputs as
+the unoptimized plan.**
+
+**Users can control the timing of the output of time series queries**.  Users
```

Comment:
Preamble so far is very long and could be shortened. It says several times that we'll be more efficient, more optimized etc, but without being clear as to what it means for the user (and what the less desirable alternative or current pain point would be).

I'd say, opening up early with the "compute on unbounded streams in bounded memory" angle would solve that.